src/DynamoCore/Models/RecordableCommands.cs

```diff
@@ -232,6 +287,69 @@ internal static RecordableCommand Deserialize(XmlElement element)
             #endregion
         }
 
+        /// <summary>
+        /// This class is base for those RecordableCommands that should have 
+        /// Guid NodeId that causes the problems during deserialization
+        /// </summary>
+        [DataContract]
+        public abstract class HasNodeIDCommand : RecordableCommand
+        {
+            internal Guid NodeId { get; private set; }
+
+            [DataMember]
+            internal string NodeIdAsString
+            {
+                get { return NodeId.ToString(); }
+                private set
+                {
+                    Guid guid;
+                    NodeId = Guid.TryParse(value, out guid) ? guid : new Guid();
+                }
+            }
+
+            protected HasNodeIDCommand(string node)
+            {
+                NodeIdAsString = node;
+            }
+
+            protected HasNodeIDCommand(Guid node)
+            {
+                NodeId = node;
+            }
+        }
+
+        /// <summary>
+        /// This class is base for those RecordableCommands that should have 
+        /// Guid ModelGuid that causes the problems during deserialization
+        /// </summary>
+        [DataContract]
```

Comment:
"that cause problems during deserialization" - Can we get to the core issue here?  @Benglin Maybe you can explain why these two commands were introduced?